### PR TITLE
New authentication

### PR DESCRIPTION
**Pull request by eljasala**
@karthikbr82 PTAL

### CODE REVIEW

File: callstats-java-sample-maven/sample.sdk/callstats-java-sdk.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CallStats.BaseURL = http://192.168.99.100:3000
+CallStats.AuthBaseURL = http://192.168.99.100:3001
+CallStats.AuthTimeOut = 5000
+CallStats.ConnectionTimeOut = 30000
+CallStats.SOTimeOut = 30000
+CallStats.keepAliveInterval = 1000
```
Comment (karthikbr82) [inline callstats-java-sample-maven/sample.sdk/callstats-java-sdk.properties]: can u change this keepalive interval from 1s to 30s

File: callstats-java-sdk/src/main/java/io/callstats/sdk/httpclient/CallStatsHttpClient.java [hunk 2]
```diff
@@ -175,12 +187,43 @@ private HttpPost generateHttpPostRequest(String url, String body) {
...
+	 * @param body the body
+	 * @return the http post
+	 */
+	private HttpPost generateHttpPostRequest(String url, UrlEncodedFormEntity form) {
+		URI uri = generateURI(url);
+
+		HttpPost post = new HttpPost(uri);
+		post.setHeader("Content-type", "application/x-www-form-urlencoded");
+	    post.setHeader("Accept", "application/json");
+		post.setEntity(form);
```
Comment (karthikbr82) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/httpclient/CallStatsHttpClient.java]: alignment.

File: callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsAuthenticator.java [hunk 3]
```diff
@@ -334,30 +200,12 @@ public void onFailure(Exception e) {
 	private void scheduleAuthentication(final int appId, final String appSecret, final String bridgeId, final CallStatsHttpClient httpClient) {
 		scheduler.schedule(new Runnable() {
```
Comment (karthikbr82) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsAuthenticator.java]: If appSecret is not used anywhere thn can we remove it from the function args?

File: callstats-java-sample-maven/sample.sdk/callstats-java-sdk.properties (hunk 1 above)
Comment (eljasala) [inline callstats-java-sample-maven/sample.sdk/callstats-java-sdk.properties]: Do you want me to change it everywhere? This is just the sample application.

**Comment by karthikbr82** [comment]
LGTM

**Pull request merged**